tests/lib/collectors/gcloud_e2e.py

```diff
@@ -0,0 +1,253 @@
+        16,
+        remote_instance_name=self.instance_to_analyse,
+    )
+
+    self.analysis_vm = gcloud_collector.analysis_vm
+    self.assertEqual(gcloud_collector.disk_names, [])
+    self.__assertCollectorIsInitialized(gcloud_collector)
+
+    # Attach the boot disk copy to the analysis VM
+    gcloud_collector.Process()
+    self.__assertInstanceAndDisk()
+
+  def test_end_to_end_other_disk(self):
+    """End to end test on GCP for the gcloud.py collector.
+
+    This end-to-end test runs directly on GCP and tests that:
+      1. The gcloud.py collector connects to the target instance and makes a
+      snapshot of the disk passed to the 'disk_names' parameter in the
+      SetUp() method.
+      2. A new disk is created from the taken snapshot.
+      3. If an analysis VM already exists, the module will attach the disk
+      copy to the VM. Otherwise, it will create a new GCP instance for
+      analysis purpose and attach the boot disk copy to it.
+    """
+
+    self.test_state = state.DFTimewolfState(config.Config)
+    gcloud_collector = gcloud.GoogleCloudCollector(self.test_state)
+    self.assertIsNotNone(gcloud_collector)
+
+    # This should make a copy of the disk specified in 'disk-names'
+    gcloud_collector.SetUp(
+        self.project_id,
+        self.project_id,
+        self.incident_id,
+        self.zone,
+        42.0,
+        16,
+        remote_instance_name=self.instance_to_analyse,
+        disk_names=self.disk_to_forensic
+    )
+
+    self.analysis_vm = gcloud_collector.analysis_vm
+    self.assertEqual(gcloud_collector.disk_names, [self.disk_to_forensic])
+    self.__assertCollectorIsInitialized(gcloud_collector)
+
+    # Attach the disk copy to the analysis VM
+    gcloud_collector.Process()
+    self.__assertInstanceAndDisk()
+
+  def __assertCollectorIsInitialized(self, gcloud_collector):
+    self.assertEqual(self.test_state.errors, [])
+    self.assertEqual(gcloud_collector.analysis_project.project_id,
+                     self.project_id)
+    self.assertEqual(gcloud_collector.remote_project.project_id,
+                     self.project_id)
+    self.assertEqual(gcloud_collector.remote_instance_name,
+                     self.instance_to_analyse)
+    self.assertEqual(gcloud_collector.all_disks, False)
+
+  def __assertInstanceAndDisk(self):
+    self.assertEqual(
+        self.test_state.output[0][0], 'gcp-forensics-vm-fake-incident-id')
+    self.assertIsInstance(self.test_state.output[0][1], gcp.GoogleComputeDisk)
+    self.assertTrue(self.test_state.output[0][1].name.startswith(
+        'incidentfake-incident-id'))
+    self.assertTrue(self.test_state.output[0][1].name.endswith('-copy'))
+    self.assertIsInstance(self.analysis_vm, gcp.GoogleComputeInstance)
+    self.assertEqual(self.analysis_vm.name, 'gcp-forensics-vm-fake-incident-id')
+    self.assertEqual(
+        self.analysis_vm.list_disks(),
+        ['gcp-forensics-vm-fake-incident-id',
+         self.test_state.output[0][1].name])
```

Comment:
I think here we should be comparing dftimewolf's internal state with the things that are actually happening in GCP.

* Check that the VM that dftimewolf asked to create actually exists in the gcp project (by calling the external API)
* Check that the disks that are being created are the ones we expect (also by calling the external API)